main.go

```diff
@@ -289,57 +327,77 @@ share the same seed as long as the indexes are different.
 
 		var seed string
 		var priv crypto.PrivKey
+		var peeringAddrs []peer.AddrInfo
+		var index int
 		var err error
 
-		credDir := os.Getenv("CREDENTIALS_DIRECTORY")
-		secretsDir := ddir
+		bitswap := cctx.Bool("bitswap")
+		dhtRouting := DHTRouting(cctx.String("dht-routing"))
+		seedPeering := cctx.Bool("seed-peering")
 
-		if len(credDir) > 0 {
-			secretsDir = credDir
-		}
+		libp2p := cctx.Bool("libp2p")
 
-		// attempt to read seed from disk
-		seedBytes, err := os.ReadFile(filepath.Join(secretsDir, "seed"))
-		if err != nil {
-			if errors.Is(err, fs.ErrNotExist) {
-				// set seed from command line or env-var
-				seed = cctx.String("seed")
-			} else {
-				return fmt.Errorf("error reading seed credentials: %w", err)
-			}
-		} else {
-			seed = strings.TrimSpace(string(seedBytes))
+		// as a convenience to the end user, and to reduce confusion
+		// libp2p is disabled when remote backends are defined
+		remoteBackends := cctx.StringSlice("remote-backends")
+		if len(remoteBackends) > 0 {
+			fmt.Printf("RAINBOW_REMOTE_BACKENDS set, forcing RAINBOW_LIBP2P=false\n")
+			libp2p = false
+			bitswap = false
+			dhtRouting = DHTOff
 		}
```

Comment:
:information_source: ended up with this, to ensure we are not running into unexpected edge cases that we missed in test coverage, or may have in future after someone refactors things.

TLDR: for end user, logic is simple: setting a remote backend effectively disables libp2p and related features, and only HTTP backend and routers remain.

End user UX is:

```console
$ RAINBOW_REMOTE_BACKENDS=http://127.0.0.1:8080 ./rainbow
Starting rainbow dev-build
RAINBOW_REMOTE_BACKENDS set, forcing RAINBOW_LIBP2P=false
```